### ISSUE TITLE
No WebProxy support under Linux - CentOS on Docker, .NET CORE 2.2

### ISSUE THREAD

**Issue by arnamq** [dotnet/wcf#3317]
Hi,

It seems that proxy support under Linux dosn't work well. I set environment variables http_proxy and https_proxy but System.ServiceModel.Channels.HttpChannelFactory ignores it and I get the exception: System.ServiceModel.CommunicationException Network is unreachable

I use CentOS under Docker with .NET CORE 2.2

I've done simple test with bare HttpClient and it works fine with a proxy.

```
 Stack Trace:    at System.ServiceModel.Channels.HttpChannelUtilities.ProcessGetResponseWebException(HttpRequestException requestException, HttpRequestMessage request, HttpAbortReason abortReason)
   at System.ServiceModel.Channels.HttpChannelFactory`1.HttpClientRequestChannel.HttpClientChannelAsyncRequest.SendRequestAsync(Message message, TimeoutHelper timeoutHelper)
   at System.ServiceModel.Channels.RequestChannel.RequestAsync(Message message, TimeSpan timeout)
   at System.ServiceModel.Channels.RequestChannel.RequestAsyncInternal(Message message, TimeSpan timeout)
   at System.Runtime.TaskHelpers.WaitForCompletionNoSpin[TResult](Task`1 task)
   at System.ServiceModel.Channels.RequestChannel.Request(Message message, TimeSpan timeout)
   at System.ServiceModel.Dispatcher.RequestChannelBinder.Request(Message message, TimeSpan timeout)
   at System.ServiceModel.Channels.ServiceChannel.Call(String action, Boolean oneway, ProxyOperationRuntime operation, Object[] ins, Object[] outs, TimeSpan timeout)
   at System.ServiceModel.Channels.ServiceChannelProxy.InvokeService(MethodCall methodCall, ProxyOperationRuntime operation)
   at System.ServiceModel.Channels.ServiceChannelProxy.Invoke(MethodInfo targetMethod, Object[] args)
--- End of stack trace from previous location where exception was thrown ---
   at System.Reflection.DispatchProxyGenerator.Invoke(Object[] args)
   at generatedProxy_1.runReportJob(ReportJob )
   at WP.Hurtownia.DFP.RunWorker.TakeReport(ReportConfiguration configuration, DateTime startDateTime, DateTime endDateTime, ClientConfiguration dfpConfiguration) in /src/WP.Hurtownia.DFP/RunWorker.cs:line 171) ---> WP.Hurtownia.DFP.Exceptions.DfpException: Problem when download report, System.ServiceModel.CommunicationExceptionNetwork is unreachable
```
Kind regards,
Piotr

**Comment by mconnew** [comment]
Are you setting any proxy related properties on your binding? What version of the WCF nuget packages are you using? With all the default values, we don't configure anything on HttpClientHandler with regards the proxy. [Here is the code](https://source.dot.net/#System.Private.ServiceModel/System/ServiceModel/Channels/HttpChannelFactory.cs,112) where we populate `HttpChannelFactory<TChannel>._proxy` or `HttpChannelFactory<TChannel>._proxyFactory`. As you can see, if `ProxyAddress` is null and `UseDefaultWebProxy` is true (which is the default value), we leave both of those fields null. [Here is the code](https://source.dot.net/#System.Private.ServiceModel/System/ServiceModel/Channels/HttpChannelFactory.cs,339) where we set `HttpClientHandler.Proxy` and `HttpClientHandler.UseProxy` and you can see we only do it when either of the two fields are not null. `HttpClientHandler.UseProxy` is set to true by default so even though we don't modify the value if no proxy properties have been set, it will still be true by default.  

As you can see, if you don't modify any proxy properties on your binding, we don't modify anything on the HttpClientHandler and you will get the same proxy behavior as a bare HttpClient.  

One possibility is that you are using an older version of WCF. We had a bug in an earlier version which broke proxy usage in some scenarios. If you are using 4.5+ version of WCF (I think it was fixed in 4.4, maybe earlier, but 4.5 is definitely good), this isn't the issue you are hitting.

**Comment by arnamq** [comment]
As you supposed the problem Was that we used version 4.4.2 of WCF. After upgradning to 4.5 everything works fine.

Than you.
Kind regards,
Piotr
